app/src/org/commcare/tasks/LogSubmissionTask.java

```diff
@@ -103,35 +104,39 @@ public static String getSubmissionUrl(SharedPreferences appPreferences) {
 
     @Override
     protected LogSubmitOutcomes doInBackground(Void... params) {
-        try {
-            SqlStorage<DeviceReportRecord> storage =
-                    CommCareApplication.instance().getUserStorage(DeviceReportRecord.class);
+        if (HiddenPreferences.isLoggingEnabled()) {

```

Comment:
Hey, the only change here is all the existing code is wrapped inside this ` if (HiddenPreferences.isLoggingEnabled()) {`, rest all are because of formatting.